storage/pinlogs_blockwise.h

```diff
@@ -0,0 +1,76 @@
+
+// The length of the PIN entry log or the PIN success log in words.
+#define PIN_LOG_WORDS 2
+
+// The length of a word in bytes.
+#define WORD_SIZE (sizeof(uint32_t))
+
+uint16_t crc16(const uint8_t *data, uint32_t length) {
+  //  poly=0x1021, init=0xFFFF, xorout=0x0000 = CRC-16/CCITT-FALSE
+  uint8_t x;
+  uint16_t crc = 0xFFFF;
+
+  while (length--) {
+    x = crc >> 8 ^ *data++;
+    x ^= x >> 4;
+    crc = (crc << 8) ^ ((uint16_t)(x << 12)) ^ ((uint16_t)(x << 5)) ^
+          ((uint16_t)x);
+  }
+  return crc;
+}
```

Comment:
I agree that it would be better to move it to trezor-crypto, but we don't have to do it in this pull request. I would just make it `static` for now.

By the way, since we have 4 bytes left in the block, we could use CRC32 instead of CRC16, if it had some benefit for us.

Reply:
made static here: https://github.com/trezor/trezor-firmware/pull/3228/commits/e280c472a3ed90e880bcb81bc017d2e3a18b14a2

we can change to crc32, or add some different redundancy (e.g. add crc32 along with the crc16, or copy the key and the counter or something?)

Reply:
> we can change to crc32, or add some different redundancy (e.g. add crc32 along with the crc16, or copy the key and the counter or something?)

All options make sense, but I am unable to decide which one could give us the best benefit.

Reply:
added a repetition of the counter into the free bytes - should be better than leaving these empty, but i don't know whats best either

https://github.com/trezor/trezor-firmware/commit/2f1e91e1d738ef021f97b9eac6834db6eea2699c

 (still 16 byte per counter although encoding it like andrew proposed make sense too).

Reply:
It looks alright.